### PR TITLE
Guardrails w Bedrock and NeMo

### PR DESCRIPTION
**Pull request by harelix**
## Amazon Bedrock and NVIDIA's NeMo Guardrails  

In this notebook, you'll engage with guardrails that have been configured using both "Custom code" and "NeMo's" configurations. These guardrails are designed to seamlessly integrate with the "faiss vector store" and "Amazon Bedrock LLM" 

Guardrails for LLMs act as control mechanisms to ensure that LLM generated responses remain within desired parameters, preventing and correcting unwanted content output. They are programmable to follow specified interaction paths, respond to certain user requests in particular ways, and maintain a designated language style, among other controls.

### CODE REVIEW

File: 09_Guardrails/vector_store/db__init_flows_index_faiss.index/index.faiss
Comment (tchattha) [inline 09_Guardrails/vector_store/db__init_flows_index_faiss.index/index.faiss:1]: why is this file being shipped with the lab, shouldn't this be created as part of the lab?

File: 09_Guardrails/vector_store/db__init_flows_index_faiss.index/index.pkl
Comment (tchattha) [inline 09_Guardrails/vector_store/db__init_flows_index_faiss.index/index.pkl:1]: why is this file being shipped with the lab, shouldn't this be created as part of the lab?

File: 09_Guardrails/vector_store/db__init_user_message_index_faiss.index/index.faiss
Comment (tchattha) [inline 09_Guardrails/vector_store/db__init_user_message_index_faiss.index/index.faiss:1]: why is this file being shipped with the lab, shouldn't this be created as part of the lab?

File: 09_Guardrails/vector_store/db__init_user_message_index_faiss.index/index.pkl
Comment (tchattha) [inline 09_Guardrails/vector_store/db__init_user_message_index_faiss.index/index.pkl:1]: why is this file being shipped with the lab, shouldn't this be created as part of the lab?

File: 09_Guardrails/vector_store/db_build_faiss.index/index.faiss
Comment (tchattha) [inline 09_Guardrails/vector_store/db_build_faiss.index/index.faiss:1]: why is this file being shipped with the lab, shouldn't this be created as part of the lab?

File: 09_Guardrails/vector_store/db_build_faiss.index/index.pkl
Comment (tchattha) [inline 09_Guardrails/vector_store/db_build_faiss.index/index.pkl:1]: why is this file being shipped with the lab, shouldn't this be created as part of the lab?

**Review by tchattha — CHANGES_REQUESTED** [review]
Please remove the pickle and vector store files, as they should not be shipped with the code, and if needed they must be generated from the code itself

File: 09_Guardrails/vector_store/db__init_bot_message_index_faiss.index/index.faiss
[the remark this answers is not in the document]
Comment (harelix) [reply 09_Guardrails/vector_store/db__init_bot_message_index_faiss.index/index.faiss:1]: faiss persistency is there to save us from repeatedly rebuilding the knowledge base and vector structure for Nemo, which helps us save time. I can remove it and let the notebook process handle the structures construction.  I will submit an update to remove the folder.

**Review by tchattha — APPROVED** [review]

File: 09_Guardrails/00_llm_guardrails_w_bedrock_nemo.ipynb
Comment (tchattha) [inline 09_Guardrails/00_llm_guardrails_w_bedrock_nemo.ipynb:1]: I am accepting this file for now, but you have to create another PR to move the installation lines from this file to the setup notebook to make it consistent with another PR.

**Review by tchattha — APPROVED** [review]
